src/utils/transformParams.js

```diff
@@ -0,0 +1,33 @@
+/* istanbul ignore file */
+import { transformStringDateToUnixTimestamp } from 'src/utils/dateTime';
+
+const transformParams = (params) =>
+  Object.keys(params).filter((param) => params[param] !== '').reduce((acc, item) => {
+    switch (item) {
+      case 'dateFrom':
+        if (params[item]) {
+          if (!acc.timestamp) acc.timestamp = ':';
+          acc.timestamp = acc.timestamp.replace(
+            /(\d+)?:/,
+            `${transformStringDateToUnixTimestamp(params[item])}:`,
+          );
+        }
+        break;
+      case 'dateTo':
+        if (params[item]) {
+          if (!acc.timestamp) acc.timestamp = ':';
+          // We add 86400 so the range is inclusive
+          acc.timestamp = acc.timestamp.replace(
+            /:(\d+)?/,
+            `:${transformStringDateToUnixTimestamp(params[item]) + 86400}`,
+          );
+        }
+        break;
+      default:
+        acc[item] = params[item];
+    }
```

Comment:
1- please avoid switch case: [more info](https://ultimatecourses.com/blog/deprecating-the-switch-statement-for-object-literals)
2- code has duplication and its hard to understand
3- care about the performance

Reply:
@soroushm This was an existing function in the blocks page. All I did was extract it to a separate function based on @eniolam1000752's recommendation